docs/tokens.md

```diff
@@ -0,0 +1,63 @@
+> contract security guarantees of a particular token. Please use any token on your own risk and judgement.
+
+## Listing a token
+
+### as user
+
+To list a new token on zkSync, go the [tokens page] (opens new window)of the explorer and click the "Add New Token"
+button and follow the instructions.
+
+### as operator
+
+1.  Deploy your ERC20 contract and take note of its _address_
+    1.  You can use `zk deploy-erc20 new <name> <symbol> <decimals> [<implementation>]` which will output
+        ```json
+        {
+          "address": "0x186E485150dDAB799Be8e25f87D717dEc45a86e5",
+          "name": "ProRif",
+          "symbol": "PRIF",
+          "decimals": "18"
+        }
+        ```
+2.  In rollup run `zk run governance-add-erc20 new <address>`. this will:
+    1. collect fees (token listing fees) on the `TokenGovernance` contract if caller is not on the `tokenLister` list
+       and `listingFee > 0` (see [deployment](/docs/contract.md#deployment) and
+       [Governance](/docs/contract.md#governance) for details)
+    2. adds token to the `Governance` contract
+    3. emits `NewToken` event
+    4. The token symbol and decimals will be fetched from our internal token list based on data from CoinGecko. If your
+       token is not on Coingecko, it will be listed as ERC20-<Token ID> with 18 decimals. Please note, that if your
+       token was listed on Coingecko after 2021-07-14, it may not be present in our token list and so make sure to
+       contact the IOVLabs RIF Rollup team before listing the token. FIXME: 💣 @RIF-ROLLUP-CORE-TEAM we could probably
+       change this behaviour
+    5. the event is captured by the `eth_watch` module in `zksync_core` and token stored in the database using
+       `token_handler` module
+
+## Fee costs
+
+In RIF Rollup the cost of every transaction has two components:
+
+Off-chain part (storage + prover costs): the cost of the state storage and the SNARK (zero-knowledge proof) generation.
+This part depends on the use of hardware resources and is therefore invariable. Our benchmarks give estimates of ~0.001
+USD per transfer. On-chain part (gas costs): for every RIF Rollup block, the validator must pay Rootstock gas to verify
+the SNARK, plus additionally ~0.4k gas per transaction to publish the state 𝛥. The on-chain part is a variable that
+depends on the current gas price in the Rootstock network. However, this part is orders of magnitude cheaper than the
+cost of normal RBTC/ERC20 transfers.
+
+## How fees are paid
+
+Transfers in RIF Rollup support "gasless meta-transactions": users pay transaction fees in the tokens being transferred.
+Thus, for example, if you want to transfer RIFPro stable-coin, there is no need for you to own RBTC or any other tokens.
+Just pay your fees in a fraction of RIFPro.
+
+Tokens listed on Coingecko with $500 of liquidity can be used to pay for fees. For tokens without a liquid market price,
```

Comment:
Is `$500` configurable? If so, how?
What about batching multiple transactions to have the possibility to pay with any tokens?

Reply:
done

Reply:
Thank you